src/gretel_synthetics/generator.py

```diff
@@ -0,0 +1,236 @@
+from typing import TYPE_CHECKING, Callable, List, Iterable, Optional, Tuple
+from dataclasses import dataclass, asdict
+from collections import namedtuple
+
+import cloudpickle
+
+import sentencepiece as spm
+import tensorflow as tf
+
+from .model import _build_sequential_model
+
+if TYPE_CHECKING:
+    from .config import BaseConfig, LocalConfig
+
+
+def _load_tokenizer(store: "LocalConfig") -> spm.SentencePieceProcessor:
+    sp = spm.SentencePieceProcessor()
+    sp.Load(store.tokenizer_model)
+    return sp
+
+
+def _prepare_model(
+    sp: spm.SentencePieceProcessor, batch_size: int, store: "LocalConfig"
```

Comment:
Can potentially avoid the `LocalConfig` forward ref here based on import logic change